holoviews/plotting/plotly/plot.py

```diff
@@ -28,6 +31,31 @@ def state(self):
         return self.handles['fig']
 
 
+    def refresh(self, **kwargs):
+        """
+        Refreshes the plot by rerendering it and then pushing
+        the updated data if the plot has an associated Comm.
+        """
+        traverse_setter(self, '_force', True)
+        key = self.current_key if self.current_key else self.keys[0]
+        dim_streams = [stream for stream in self.streams
+                       if any(c in self.dimensions for c in stream.contents)]
+        stream_params = stream_parameters(dim_streams)
+        key = tuple(None if d in stream_params else k
+                    for d, k in zip(self.dimensions, key))
+        stream_key = wrap_tuple_streams(key, self.dimensions, self.streams)
+
+        # Update if not top-level, batched or an ElementPlot
+        if self.top_level:
+            self.update(stream_key)
+        else:
+            self.current_key = None
+            self.current_frame = None
```

Comment:
Should probably just factor this bit out. I'll think about it and probably clean this up before merging.